### PR TITLE
Fixing  save as and quit and adding some new methods 

### PR DESCRIPTION
**Pull request by kiok46**
As I was going through the code I saw couple of bugs, so decided to fix them and made little changes. 

* Added a new confirmation dialog box for the Quit button. (see picture) 
* Added 2 new object properties that might also be helpful in future. 
* Added a new method for "quit" action button to handle.
* And also there was a bug related to save as.

screenshot1

### CODE REVIEW

File: designer/app.py [hunk 1]
```diff
@@ -135,6 +136,17 @@ class Designer(FloatLayout):
        a :class:`~kivy.properties.BooleanProperty`
     '''
 
+    _quit_on_save = BooleanProperty(False)
```
Comment (aron-bordin) [inline designer/app.py]: we should avoid creating this new variable. It's a variable to create a kind of restriction, let's keep it clean. See the comments below why it's not necessary

File: designer/designer.kv [hunk 2]
```diff
@@ -1649,7 +1668,7 @@
                     on_release: root.action_btn_settings_pressed()
                 DesignerActionButton:
                     text: 'Quit'
-                    on_release: root.action_btn_quit_pressed()
+                    on_release: root.action_btn_quit_check()
```
Comment (aron-bordin) [inline designer/designer.kv]: you can use the `action_btn_quit_pressed`

**Comment by aron-bordin** [comment]
I think that it's better to continue using this old method: https://github.com/kivy/kivy-designer/blob/master/designer/app.py#L1130

Instead of creating a new method to treat an unsaved new project, I think that it's better to use just one method to treat any kind of unsaved project. It's simpler.

Check that we have a duplicated method: https://github.com/kivy/kivy-designer/blob/master/designer/app.py#L1111 and https://github.com/kivy/kivy-designer/blob/master/designer/app.py#L1130

You can edit these method to call a new  "check_quit" method.
`check_quit` will be very similar with "action_btn_quit_pressed_new_project" that you have created.

If there is something unclear, just let me know. I'm just trying to move your code to different positions to keep it simpler and to be able to use your new dialog in new or already created projects

**Comment by kiok46** [comment]
That's a good thought :) , it's better to use just one method to treat any kind of unsaved project.
 OK I will improve it.

File: designer/app.py [hunk 3]
```diff
@@ -1115,17 +1142,8 @@ def on_request_close(self, *args):
...
-                                               ' lost.\n'
-                                               'Do you want to quit?')
-        self._confirm_dlg.bind(on_ok=self._perform_quit,
-                               on_cancel=self._cancel_popup)
 
-        self._popup = Popup(title='Quit', content=self._confirm_dlg,
-                            size_hint=(None, None), size=('200pt', '150pt'),
-                            auto_dismiss=False)
-        self._popup.open()
-        return True
```
Comment (aron-bordin) [inline designer/app.py]: just keep this return True, otherwise the window will be closed.
Your pr is really great, just fix this line so we can merge

**Comment by aron-bordin** [comment]
@kiok46 great work, thx :+1:

**Pull request merged**